packages/flutter/lib/src/cupertino/text_field.dart

```diff
@@ -0,0 +1,600 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/rendering.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter/widgets.dart';
+
+import 'colors.dart';
+import 'icons.dart';
+import 'text_selection.dart';
+
+export 'package:flutter/services.dart' show TextInputType, TextInputAction, TextCapitalization;
+
+const BorderSide _kDefaultRoundedBorderSide = BorderSide(
+  color: CupertinoColors.lightBackgroundGray,
+  style: BorderStyle.solid,
+  width: 0.0,
+);
+const Border _kDefaultRoundedBorder = Border(
+  top: _kDefaultRoundedBorderSide,
+  bottom: _kDefaultRoundedBorderSide,
+  left: _kDefaultRoundedBorderSide,
+  right: _kDefaultRoundedBorderSide,
+);
+const BoxDecoration _kDefaultRoundedBorderDecoration = BoxDecoration(
+  border: _kDefaultRoundedBorder,
+  borderRadius: BorderRadius.all(Radius.circular(4.0)),
+);
+
+const TextStyle _kDefaultTextStyle = TextStyle(
+  fontFamily: '.SF Pro Text',
+  fontSize: 17.0,
+  letterSpacing: -0.38,
+  color: CupertinoColors.black,
+  decoration: TextDecoration.none,
+);
+
+const Color _kSelectionHighlightColor = Color(0x667FAACF);
```

Comment:
A thought on all of these constants we have throughout cupertino files...I'm thinking that maybe we should aggregate all such constants in a single cupertino file so that other developers can use them, if desired.  I'm sure developers will need to create their own iOS styled widgets and it would be nice if they had a way to ensure they're using the same colors, border radii, etc. that we use.

Reply:
I agree. Common ones should be in cupertino/colors.dart. Though iOS doesn't really semantically catalog them in a public place so I hesitate to put new ones in there each time I encounter a new color since namespace real estate is valuable. I generally comb the code each time I see a color and if I can figure out a pattern, I add it to CupertinoColors. For the ones that are seemingly domain specific like these ones as far as I can tell for now, I leave them private in their place of use.